plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1881,6 +1914,71 @@ public void renameTable(ConnectorSession session, ConnectorTableHandle tableHand
         metastore.renameTable(session, handle.getSchemaTableName(), newTableName);
     }
 
+    @Override
+    public void setTableProperties(ConnectorSession session, ConnectorTableHandle tableHandle, Map<String, Optional<Object>> properties)
+    {
+        DeltaLakeTableHandle handle = (DeltaLakeTableHandle) tableHandle;
+        ConnectorTableMetadata tableMetadata = getTableMetadata(session, handle);
+        ProtocolEntry currentProtocolEntry = getProtocolEntry(session, handle.getSchemaTableName());
+
+        Optional<Integer> readerVersion = Optional.empty();
+        if (properties.containsKey(READER_VERSION_PROPERTY)) {
+            readerVersion = Optional.of((int) properties.get(READER_VERSION_PROPERTY)
+                    .orElseThrow(() -> new IllegalArgumentException("The " + READER_VERSION_PROPERTY + " property cannot be empty")));
+            if (readerVersion.get() < currentProtocolEntry.getMinReaderVersion()) {
+                throw new TrinoException(DELTA_LAKE_BAD_DATA, format(
+                        "%s cannot be downgraded from %d to %d", READER_VERSION_PROPERTY, currentProtocolEntry.getMinReaderVersion(), readerVersion.get()));
+            }
+        }
+
+        Optional<Integer> writerVersion = Optional.empty();
+        if (properties.containsKey(WRITER_VERSION_PROPERTY)) {
+            writerVersion = Optional.of((int) properties.get(WRITER_VERSION_PROPERTY)
+                    .orElseThrow(() -> new IllegalArgumentException("The " + WRITER_VERSION_PROPERTY + " property cannot be empty")));
+            if (writerVersion.get() < currentProtocolEntry.getMinWriterVersion()) {
+                throw new TrinoException(DELTA_LAKE_BAD_DATA, format(
+                        "%s cannot be downgraded from %d to %d", WRITER_VERSION_PROPERTY, currentProtocolEntry.getMinWriterVersion(), writerVersion.get()));
+            }
+        }
+
+        long commitVersion = handle.getReadVersion() + 1;
+
+        List<String> partitionColumns = getPartitionedBy(tableMetadata.getProperties());
+        List<DeltaLakeColumnHandle> columns = tableMetadata.getColumns().stream()
+                .filter(column -> !column.isHidden())
+                .map(column -> toColumnHandle(column, column.getName(), column.getType(), partitionColumns))
+                .collect(toImmutableList());
+        ProtocolEntry protocolEntry = new ProtocolEntry(
+                readerVersion.orElse(currentProtocolEntry.getMinReaderVersion()),
+                writerVersion.orElse(currentProtocolEntry.getMinWriterVersion()));
+
+        try {
+            TransactionLogWriter transactionLogWriter = transactionLogWriterFactory.newWriter(session, handle.getLocation());
+
+            appendTableEntries(
```

Comment:
`appendTableEntries` also appends a `metadata` entry, but we could just get away with `commitInfo` and `protocol`, I think